awx/api/views/__init__.py

```diff
@@ -2568,10 +2568,34 @@ def _validate_spec_data(new_spec, old_spec):
                         return Response(dict(error=_(
                             "The {min_or_max} limit in survey question {idx} expected to be integer."
                         ).format(min_or_max=key, **context)))
-            if qtype in ['multiplechoice', 'multiselect'] and 'choices' not in survey_item:
-                return Response(dict(error=_(
-                    "Survey question {idx} of type {survey_item[type]} must specify choices.".format(**context)
-                )))
+            # if it's a multiselect or multiple choice, it must have coices listed
+            # choices and defualts must come in as strings seperated by /n characters.
+            if qtype == 'multiselect' or qtype == 'multiplechoice':
+                if 'choices' in survey_item:
+                    if isinstance(survey_item['choices'], str):
+                        survey_item['choices'] = '\n'.join(choice for choice in survey_item['choices'].splitlines() if choice.strip() != '')
+                else:
+                    return Response(dict(error=_(
+                        "Survey question {idx} of type {survey_item[type]} must specify choices.".format(**context)
+                    )))
+                # If there is a default, split it out removing extra /n characters.
+                # This should happen for both multiple choice and multiselect, even though multiple choice should only have one default.
+                # There can still be extra newline characters added in the API.
+                if 'default' in survey_item:
+                    if isinstance(survey_item['default'], str):
+                        survey_item['default'] = '\n'.join(choice for choice in survey_item['default'].splitlines() if choice.strip() != '')
+                        list_of_defaults = survey_item['default'].splitlines()
+                    else:
+                        list_of_defaults = [survey_item('default')]
```

Comment:
I don't get this. Problem 1 is `survey_item('default')` vs `survey_item['default']`. Then, if it's a list, it should be used as-is instead of putting in a new list.

Reply:
no, other case converts str-to-str. If you get a list, then you might either do list-to-str or leave it alone.

Reply:
You're 100% right. I think I'll leave it alone if it's a list since I'll be traversing it and I split the string in the above block into a list using splitlines().